src/main/java/coresearch/cvurl/io/mapper/BodyType.java

```diff
@@ -0,0 +1,32 @@
+package coresearch.cvurl.io.mapper;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+/**
+ * Class that describes type with generics. Used as parameter to some of {@link coresearch.cvurl.io.request.Request}
+ * methods to provide ability to parse response to body to class with generics.
+ * Usage example: <br/>
+ * {@code cvurl.get(url).asObject(new BodyType<List<Users>() {});} <br/>
+ * this code snippet would parse response body to {@code List<User>}
+ */
+public abstract class BodyType<T> {
+    private Type type;
+
+    protected BodyType() {
+        Type superclass = this.getClass().getGenericSuperclass();
```

Comment:
It will be better to cast to (ParameterizedType) here. In one place

Reply:
yap, you are right, gonna change this